src/lib.rs

```diff
@@ -131,51 +119,39 @@ fn print_message(buf: &[u8]) {
     println!();
 }
 
-fn spawn_cmd_sender<'f>(
-    local_ip: IpAddr,
-    target_ip: IpAddr,
-    port: u16,
-) -> AnyResult<Sender<Vec<u8>>> {
-    let socket = UdpSocket::bind((local_ip, port)).map_err(|e| {
-        anyhow!(
-            "Couldn't bind to local socket {}:{} - {}",
-            local_ip,
-            port,
-            e
-        )
-    })?;
+fn spawn_cmd_sender(local_ip: IpAddr, target_addr: SocketAddr) -> AnyResult<Sender<Vec<u8>>> {
+    let local_addr = SocketAddr::new(local_ip, target_addr.port());
+
+    let socket = UdpSocket::bind(local_addr)
+        .map_err(|e| anyhow!("Couldn't bind to local socket {} - {}", local_addr, e))?;
 
     let (tx, rx) = channel::<Vec<u8>>();
     std::thread::spawn(move || loop {
-        let mut frame_to_send = rx.recv().expect("couldn't receive frame.");
+        let frame_to_send = rx.recv().expect("couldn't receive frame.");
 
         print_message(&frame_to_send);
         let size = socket
-            .send_to(&mut frame_to_send, (target_ip, port))
+            .send_to(&frame_to_send, target_addr)
             .expect("something terrible happened");
         println!("sent {}", size);
     });
     Ok(tx)
 }
 
 impl Drone {
-    pub fn new(addr: Option<IpAddr>) -> AnyResult<Self> {
-        let addr = addr.unwrap_or(
-            DEFAULT_ADDR
-                .parse()
-                .expect("couldn't parse default ip address"),
-        );
-
+    pub fn new(addr: IpAddr) -> AnyResult<Self> {
```

Comment:
could we maybe add a default() impl that calls new() with the default drone ip ? (192.168.2.1)

Reply:
I was thinking about that - it really depends on the local setup, I think we can have the static for the Jumping sumo?

Reply:
Indeed ! It might be specific to the type of drone so it’s a great idea !